source/extensions/config_subscription/grpc/xds_mux/delta_subscription_state.cc

```diff
@@ -188,8 +190,24 @@ void DeltaSubscriptionState::handleGoodResponse(
     }
   }
 
-  absl::Span<const envoy::service::discovery::v3::Resource* const> non_heartbeat_resources_span =
-      absl::MakeConstSpan(non_heartbeat_resources.data(), non_heartbeat_resources.size());
+  absl::Span<const envoy::service::discovery::v3::Resource* const> non_heartbeat_resources_span;
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.xds_prevent_resource_copy")) {
+    // Reorder the resources in the response, having all the non-heartbeat
+    // resources at the front of the list. Note that although there's no
+    // requirement to keep stable ordering, we do so to process the resources in
+    // the order they were sent.
+    auto last_non_heartbeat = std::stable_partition(
```

Comment:
would two discovery responses with different order of resources but same resources considered equal?

Reply:
The protocol assumes that the given repeated field will be a set of resources, which is "enforced" by the having different resource names for the resources.
The implementation (in Envoy) "allows" having the "same resource" in the discovery-response, but will only take the first element in the list (e.g., [cds](https://github.com/envoyproxy/envoy/blob/312e01def6594155127d6d9a4e1a1de432551390/source/common/upstream/cds_api_helper.cc#L43-L48)).
In this PR I've attempted to keep the same order is done before the change by using a stable-partition.

Reply:
thanks for explanation!